### ISSUE TITLE
import() expressions are not external module indicators

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#22863]
```ts
// @filename: globble.ts
declare module "lol" {
  export let a: any, b: any, c: any;
}

// @filename: a.ts
let x = import("lol");

// @filename: b.ts
let x = 10;
```

**Expected**: No errors, `a.ts` is a module.
**Actual**: Errors on the type of `x`, errors on `x` being a block scoped declaration.

**Comment by weswigham** [comment]
Node 10 is slated allow top-level `import()` in cjs, soooooo....... it's not?

**Comment by DanielRosenwasser** [comment]
What about `import.meta`?

**Comment by mhegazy** [comment]
a.ts is not a module. `import` is specifically added to allow for importing a module from a script.

**Comment by mhegazy** [comment]
I would say `import.meta` is not a marker; i.e. it does not make a script file a module, but it is an error to use `import.meta` outside a module.

**Comment by Kovensky** [comment]
The original idea of unambiguous grammar is to treat as module things that are not parseable as script, so `import.meta` probably should be considered a "marker" as long as you implement unambiguous grammar.
